walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,179 @@
 
 package com.google.sps;
 
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.ListIterator;
 
 public final class FindMeetingQuery {
+  private int meetingDuration;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    // The total (common) free time we have to conduct the meeting.
+    LinkedList<TimeRange> freeTime = new LinkedList<TimeRange>();
+    freeTime.addAll(Arrays.asList(TimeRange.WHOLE_DAY));
+
+    meetingDuration = (int) request.getDuration();
+    Collection<String> optionalAttendees = request.getOptionalAttendees();
+    Collection<String> mandatoryAttendees = request.getAttendees();
+
+    // If duration of the meeting is greater than a whole day, then return an empty list.
+    if (meetingDuration > TimeRange.WHOLE_DAY.duration()) {
+      return Arrays.asList();
+    }
+
+    // If there are no attendees then return the whole day (ie the free time we have till now).
+    if (mandatoryAttendees.isEmpty() && optionalAttendees.isEmpty()) {
+      return freeTime;
+    }
+
+    // To store all the events which the optional attendees are attending.
+    LinkedList<Event> optionalAttendeeEvents = new LinkedList<>();
+
+    /* Changes freeTime such that it now conatins all the free time slots when the meeting can
```

Comment:
It's better to use "//" for non-Javadoc comments.

Reply:
Okay